regression_tests/tests/gitbook/test_docs.py

```diff
@@ -0,0 +1,181 @@
+import os
+import re
+import argparse
+import subprocess
+from glob import glob
+from pathlib import Path
+from typing import NamedTuple, List
+
+class SnippetExec(NamedTuple):
+    # The result from executing the code snippets sequentially.
+    # Returned from `run` and stores useful information for debugging that is displayed if a file fails to run to the end successfully.
+    
+    # All the Python code block in the Markdown file joined in order of appearance.
+    snippet: str
+    # Whether or not every snippet is successfully executed.
+    success: bool
+    # The output from running the snippet
```

Comment:
Is this whatever is logged to stdout? Not entirely sure what output of a code snippet necessarily is.